### PR TITLE
Now batching data in the data transfer demo to increase throughput.

### PR DESCRIPTION
**Pull request by phoglund**
This version sends data 3x faster than the original. The progress bars are a bit jankier as a result but I think throughput is more important.

### CODE REVIEW

File: src/content/datachannel/datatransfer/js/main.js [hunk 1]
```diff
@@ -157,7 +167,7 @@ function onReceiveMessageCallback(event) {
   receivedSize += event.data.length;
   receiveProgress.value = receivedSize;
 
-  if (receivedSize >= bytesToSend) {
+  if (receivedSize == bytesToSend) {
```
Comment (samdutton) [inline src/content/datachannel/datatransfer/js/main.js]: ===

File: src/content/datachannel/datatransfer/js/main.js [hunk 2]
```diff
@@ -78,16 +78,26 @@ function randomAsciiString(length) {
...
+    // throughput quite a bit (setTimeout(fn, 0) can take hundreds of milli-
+    // seconds to execute).
+    while (sendProgress.value < sendProgress.max) {
+      if (sendChannel.bufferedAmount > 5 * chunkSize) {
+        setTimeout(sendAllData, 250);
+        return;
+      }
+      sendProgress.value += chunkSize;
+      sendChannel.send(stringToSendRepeatedly);
+    };
```
Comment (samdutton) [inline src/content/datachannel/datatransfer/js/main.js]: Remove unnecessary semi-colon.

Comment (phoglund) [inline src/content/datachannel/datatransfer/js/main.js]: Done. Note to self: run grunt locally :)

**Comment by phoglund** [comment]
PTAL

**Comment by samdutton** [comment]
LGTM.

**Comment by samdutton** [comment]
Re Grunt: I ALWAYS forget...

I've just done a PR to incorporate grunt-githooks.

Really simple to use: forces grunt on git commit.

I'll email webrtc-eng with details.

**Comment by kawa-** [comment]
Great! By the way, I have a question.

~~~
+ if (sendChannel.bufferedAmount > 5 * chunkSize) {
+ setTimeout(sendAllData, 250);
+ return;
+ }
~~~

Is there reasons for `sendChannel.bufferedAmount > 5 * chunkSize`? why five?

**Comment by samdutton** [comment]
>
> Is there reasons for sendChannel.bufferedAmount > 5 * chunkSize? why five?
>

Whatever the reason, we should make sure to comment and/or use constants
for values like this.

**Comment by phoglund** [comment]
re: Kawa: I think I did that. I seemed to get maximum throughput if I kept pumping data into the buffer, until it reached 5 chunks. It's possible higher values will work but probably not less.

Yes, documenting the above would have been nice :)

**Comment by kawa-** [comment]
@phoglund Thanks, I understand. I like heuristic ways :+1: 
@samdutton I agree :+1:

**Pull request merged**